File: test/test_local.py
# import <
from source.local import (
   
   fileSet,
   fileGet,
   fileDel
   
)

# >


def test_fileSet():
   ''' fileSet() Test Suite '''

   # test <
   result = fileSet(
      
      pFile = 'test/test.json',
      pData = {"json" : "example"}
      
   )
   assert (result == None), 'Can establish .json files with data.'
   
   result = fileSet(
      
      pFile = 'test/test.txt',
      pData = "this is an example"
      
   )
   assert (result == None), 'Can establish .txt files with data.'

   expected = 'Invalid data for this filetype.'
   result = fileSet(
      
      pFile = 'test/test.txt',
      pData = {"this" : "that"}
      
   )
   assert (result == expected), 'Can detect invalid data for .txt files.'
   
   expected = 'File already exists.'
   result = fileSet(
      
      pOverride = False,
      pFile = 'test/test.txt',
      pData = 'this is an example'
      
   )
   assert (result == expected), 'Can prevent overwritten existing files.'
   
   # >
   
   # deconstruct <
   fileDel(pFile = 'test/test.json')
   fileDel(pFile = 'test/test.txt')
   
   # >


def test_fileGet():
   ''' fileGet() Test Suite '''
   
   # construct <
   fileSet(
      
      pFile = 'test/test.txt',
      pData = 'this is a test'
      
   )
   fileSet(
      
      pFile = 'test/test.json',
      pData = {'json' : 'example'}
      
   )
   
   # >

   # test <
   expected = {"json" : "example"}
   result = fileGet(pFile = 'test/test.json')
   assert (result == expected), 'Can load existing .json files.'
      
   expected = 'this is a test'
   result = fileGet(pFile = 'test/test.txt')
   assert (result == expected), 'Can load existing .txt files.'
   
   expected = 'Loaded data is broken.'
   result = fileGet(pFile = 'test/broken.json')
   assert (result == expected), 'Can detect broken .json data.'
   
   expected = 'File does not exist.'
   result = fileGet(pFile = 'test/broken.txt')
   assert (result == expected), 'Can detect non-existing files.'
   
   # >
   
   # deconstruct <
   fileDel(pFile = 'test/test.json')
   fileDel(pFile = 'test/test.txt')
   
   # >


def test_fileDel():
   ''' fileDel() Test Suite '''
   
   # construct <
   fileSet(
      
      pFile = 'test/test.json',
      pData = {'json' : 'test'}
      
   )
   fileSet(
      
      pData = 'txt test',
      pFile = 'test/test.txt'
      
   )
   
   # >
   
   # test <
   result = fileDel(pFile = 'test/test.json')
   assert (result == None), 'Can delete existing .json files.'
   
   result = fileDel(pFile = 'test/test.txt')
   assert (result == None), 'Can delete existing .txt files.'
   
   expected = 'File does not exist.'
   result = fileDel(pFile = 'test/broken.txt')
   assert (result == expected), 'Can detect non-existing files.'
   
   # >

File: test/test_local.js
// import <
const assert = require('assert');

const {

   fileSet,
   fileGet,
   fileDel,
   dirSet,
   dirGet,
   dirDel

} = require('../source/fp/local.js');

// >


// declare <
var gTestDir = 'testdir';
var gFileTXT = 'test.txt';
var gFileJSON = 'test.json';

// >


describe('fileSet() Test Suite', async () => {

   it('can write to TXT files', async () => {

      const result = await fileSet({

         pData : 'test',
         pFile : gFileTXT

      });
      assert.strictEqual(result, undefined);

   });


   it('can write to JSON files', async () => {

      const result = await fileSet({

         pFile : gFileJSON,
         pData : {"test" : "test"}

      });
      assert.strictEqual(result, undefined);

   });


   it('can prevent writing broken data.', async () => {

      const expected = 'Could not write to file.';
      const result = await fileSet({

         pFile : gFileJSON,
         pData : undefined

      });
      assert.strictEqual(result, expected);

   });

});


describe('fileGet() Test Suite', async () => {

   it('can read TXT files', async () => {

      const expected = 'test';
      const result = await fileGet({pFile : gFileTXT});
      assert.strictEqual(result, expected);

   });


   it('can read JSON files', async () => {

      const expected = {'test' : 'test'};
      const result = await fileGet({pFile : gFileJSON});
      assert.deepEqual(result, expected);

   });


   it('can detect invalid filepaths', async () => {

      const expected = 'Could not read file.';
      const result = await fileGet({pFile : 'dne.txt'});
      assert.strictEqual(result, expected);

   });

});


describe('fileDel() Test Suite', async () => {

   it('can delete TXT files', async () => {

      const result = await fileDel({pFile : gFileTXT});
      assert.strictEqual(result, undefined);

   });


   it('can delete JSON files', async() => {

      const result = await fileDel({pFile : gFileJSON});
      assert.strictEqual(result, undefined);

   });


   it('can detect invalid files', async () => {

      const expected = 'File does not exist.';
      const result = await fileDel({pFile : 'dne.txt'})
      assert.strictEqual(result, expected);

   });

});


describe('dirSet() Test Suite', async () => {

   it('can create directories', async () => {

      const result = await dirSet({pDir : gTestDir});
      assert.strictEqual(result, undefined);

   });


   it('can detect invalid directories', async () => {

      const expected = 'Could not create directory.';
      const result = await dirSet({pDir : 'dne/dne/dne'});
      assert.strictEqual(result, expected);

   });

});


// describe('dirGet() Test Suite', async () => {

//    it('can read valid directories', async () => {

//       const expected = [

//          'test_local.js',
//          'test_remote.js'

//       ];
//       const result = await dirGet({pDir : '/test'});
//       assert.deepStrictEqual(result, expected);
      
//    });


//    it('can detect invalid directories', async () => {

//       const expected = 'Directory does not exist.';
//       const result = await dirGet({pDir : '/dne'});
//       assert.strictEqual(result, expected);

//    });

// });


// describe('dirDel() Test Suite', async () => {

//    it('can delete valid directories', async () => {

//       const result = await dirDel({pDir : gTestDir});
//       assert.strictEqual(result, undefined);

//    });

   
//    it('can detect invalid directories', async () => {

//       const expected = 'Directory does not exist.';
//       const result = await dirDel({pDir : 'dne'});
//       assert.strictEqual(result, expected);

//    });

// });